### PR TITLE
allow resizing rect

### PR DESCRIPTION
**Pull request by anshul01sharma**
Rect with edit icon at the top left, once enters edit mode with appropriate UI dragged to resize & edit icon replace with OK button.

### CODE REVIEW

File: lib/utils/resizable.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  @override
+  _ResizableState createState() => _ResizableState();
+}
+
+class _ResizableState extends State<Resizable> {
+  double initX = 300;
+  double initY = 100;
+
+  // handle the gesture of pan start
```
Comment (danikenan) [inline lib/utils/resizable.dart:20]: use better names so no need for comments

File: lib/utils/resizable.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import 'package:flutter/material.dart';
+
+// Radius of the resizable circle position
+const ballDiameter = 30.0;
+
+// Resizable class used for the dragging
```
Comment (danikenan) [inline lib/utils/resizable.dart:6]: Rename `Resizable` -> `ResizeHandle` and no need for comments

File: lib/utils/resizable.dart [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+class Resizable extends StatefulWidget {
+  const Resizable({Key? key, required this.onDrag}) : super(key: key);
+
+  final Function onDrag;
```
Comment (danikenan) [inline lib/utils/resizable.dart:10]: private with `_` 
also add safer typing

File: lib/widgets/camera_view.dart [hunk 4]
```diff
@@ -5,6 +5,7 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:google_ml_kit/google_ml_kit.dart';
 import 'package:magic_reviews_app/utils/focusRect.dart';
+import 'package:magic_reviews_app/utils/resizable.dart';
```
Comment (danikenan) [inline lib/widgets/camera_view.dart:8]: use local relative path for local package

(yes... the line above too)

File: lib/widgets/camera_view.dart [hunk 5]
```diff
@@ -30,12 +31,36 @@ class _CameraViewState extends State<CameraView> {
   // double _scale = 1.0;
   late double _maxScale, _minScale, _scale, _prevScale;
 
+  late double height, width, top, left;
```
Comment (danikenan) [inline lib/widgets/camera_view.dart:34]: unclear from naming that relates to focus rect

consider even extracting functionality of focus rect to its own widget

File: lib/widgets/camera_view.dart [hunk 6]
```diff
@@ -30,12 +31,36 @@ class _CameraViewState extends State<CameraView> {
...
+
+  bool isEdit = false;
```
Comment (danikenan) [inline lib/widgets/camera_view.dart:36]: same

File: lib/widgets/camera_view.dart [hunk 7]
```diff
@@ -153,18 +178,214 @@ class _CameraViewState extends State<CameraView> {
           children: <Widget>[
             CameraPreview(_controller!),
             if (widget.customPaint != null) widget.customPaint!,
+
+            // Edit & Ok button as per the isEdit value
```
Comment (danikenan) [inline lib/widgets/camera_view.dart:182]: no comments. extract to build method/widget if unclear

File: lib/widgets/camera_view.dart [hunk 8]
```diff
@@ -153,18 +178,214 @@ class _CameraViewState extends State<CameraView> {
...
-            )
+                constraints: const BoxConstraints.expand(),
+                decoration: BoxDecoration(
+                  color: Colors.white.withOpacity(0.3),
+                  borderRadius: BorderRadius.circular(20),
+                ),
+              ),
+            ),
+
+            // top left
```
Comment (danikenan) [inline lib/widgets/camera_view.dart:230]: extract to widget

**Review by danikenan — CHANGES_REQUESTED** [review]
overall i am pleased. we are good. but we need some improvements.

pls extract the rectangle to its own widget (or set of widgets) and apply some naming conventions as well

whenever you feel like adding a comment, stop and ensure naming and code is properly named and partitioned.

File: lib/utils/resizable.dart (hunk 1 above)
Comment (anshul01sharma) [reply lib/utils/resizable.dart:20]: Okay, I will make the proper naming conventions.

File: lib/utils/resizable.dart (hunk 3 above)
Comment (anshul01sharma) [reply lib/utils/resizable.dart:10]: regarding the return function callback it not allow underscore like this "_onDrag".

**Pull request merged**